backend/de.metas.manufacturing/src/main/java/org/eevolution/productioncandidate/service/ResourcePlanningPrecision.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * #%L
+ * de.metas.manufacturing
+ * %%
+ * Copyright (C) 2025 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package org.eevolution.productioncandidate.service;
+
+import de.metas.util.Check;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NonNull;
+import org.compiere.util.TimeUtil;
+
+import javax.annotation.Nullable;
+import java.sql.Timestamp;
+import java.util.Arrays;
+
+import static org.compiere.util.TimeUtil.TRUNC_15M;
+import static org.compiere.util.TimeUtil.TRUNC_MINUTE;
+
+@AllArgsConstructor
+public enum ResourcePlanningPrecision
+{
+	MINUTE("M"),
+	MINUTE_15("15M");
+
+	@Getter
+	private final String code;
+
+	@NonNull
+	public static ResourcePlanningPrecision ofCodeOrMinute(@Nullable final String code)
+	{
+		if (Check.isBlank(code))
```

Comment:
codeNorm = StringUtils.trimBlankToNull(code);

if(codeNorm == null)
{
return MINUTE;
}
else if(codeNorm.equals(MINUTE.code) {
return MINUTE;
}
...........
else throw

Reply:
basically, 
* guard against the case when we have sysconfig value like `M      `
* either use a `Map<String, ResourcePlanningPrecision> byCode` either go with switch/ifs. IMHO streams are not feasible for this case